guides/leon/vsftpd.md

```diff
@@ -0,0 +1,211 @@
+    ```
+    You should see something like this
+    ![3](/blog/content/images/2018/02/3.PNG)
+    
+    Now make the cert request
+    ```bash
+    openssl req -new -key FTP.key -out certificate.csr
+    ```
+    You should see this
+    ![4-1](/blog/content/images/2018/02/4-1.PNG)
+    
+    Next, lets get rid of the pass phrase on the key
+    ```bash
+    cp FTP.key original.key
+    openssl rsa -in original.key -out ftp.key
+    ```
+    
+    Now make the actual certificate (all on one line)
+    ```bash
+    openssl x509 -req -days 365 -in certificate.csr -signkey ftp.key -out my_cert.crt
+    ```
+    You should see this
+    ![5-1](/blog/content/images/2018/02/5-1.PNG)
+    ```
+    # and lastly move them to a safe plave
+    cp ftp.key /etc/ssl/private/
+    cp my_cert /etc/ssl/certs/
+    ```
+2. Lets edit the config files
+
+    Open up `vsftpd.conf` with `vi`
+    ```bash
+    # Change this line
+    pam_service_name=ftp
+    
+    # Add the following lines at the bottom
+    ssl_enable=YES
+    ssl_tlsv1=YES
+    ssl_sslv2=NO
+    ssl_sslv3=NO
+    rsa_cert_file=/etc/ssl/certs/my_cert.crt
+    rsa_private_key_file=/etc/ssl/private/ftp.key
+    ssl_ciphers=HIGH
+    ```
+    
+    Now save and restart the service
+    ```bash
+    systemctl restart vsftpd
+    ```
+    
+3. Now go test it out, [here](#test-secure)
+<br>
+
+<a id="users"></a>
+## Users
+How to add users to vsftpd you ask? This is how.
+
+1. Create a new user
+
+    ```bash
+    useradd -m jeff
+    passwd jeff
+    ```
+Thats it theres no step 2, once they are users on the system you can connect with FTP using their credentials as shown in the testing section below.
+    
+<br>
+    
+<a id="testing"></a>
+## Testing 
+Lets make sure everything works
+
+<a id="test-unsecure"></a>
+1. Test localhost - insecure
+
+    ```bash
+    # install FTP package - our iso has it pre-installed
+    apt-get install ftp
+    # now try to connect
+    ftp localhost
+    # Enter your user name
+    Name (localhost:{user-name}): {username}
+    # Enter password
+    Password: {secret-password}
+    ```
+    
+    You should now see something like so
+    ![2-1](/blog/content/images/2018/02/2-1.PNG)
+    
+<a id="test-secure"></a>
+
+2. Test localhost - securely
+
+    First try to login using the process in step 1. You should get an error like so
+    ![7](/blog/content/images/2018/02/7.PNG)
+    
+    Next lets try to login securely
+    ```bash
+    # First install lftp
+    apt-get install lftp
+    # run these commands
+    lftp
+    lftp:~> set ssl:verify-certificate no
+    lftp:~> connect localhost
+    lftp user@localhost:~> login {user}
+    Password: {password}
+    # Now lets move arround to see that it works
+    lftp user@localhost:~> cd /
+    cd ok, cwd=/
+    lftp user@localhost:~> ls
+    bin
+    root
+    ...
+    ```
+    Like so...
+    ![8](/blog/content/images/2018/02/8.PNG)
+<br>
+
+<a id="useful-commands"></a>
+## Useful Commands

```

Comment:
This is good. We have a another section called knowledge. Maybe write a cheat sheet for it. Jon is doing one for BASH.